src/python/pants/engine/isolated_process.py

```diff
@@ -86,11 +87,61 @@ def _verify_env_is_dict(cls, env):
 
 
 class ExecuteProcessResult(datatype(['stdout', 'stderr', 'exit_code', 'output_directory_digest'])):
-  pass
+  """Result of executing a process. Will raise an exception if the exit code is non-zero."""
```

Comment:
The raising behaviour is a property of the rule which generates it, rather than the generated type, right? Maybe drop this sentence, and just leave the one on the @rule

Reply:
Hm. I'm not sure. Since this is the entry point for using the rule, and the rule isn't directly reference-able I feel like it makes sense for this to be here.